pkg/kubectl/cmd/logs.go

```diff
@@ -199,7 +225,40 @@ func (o LogsOptions) Validate() error {
 
 // RunLogs retrieves a pod log
 func (o LogsOptions) RunLogs() (int64, error) {
-	req, err := o.LogsForObject(o.Object, o.Options)
+	switch t := o.Object.(type) {
```

Comment:
Ok, you lost me again. It might be easier for you do just show me what you want rather than me guess. I guess I don't see how it will all work in the end.